zcash_primitives/src/zip32.rs

```diff
@@ -134,6 +134,20 @@ impl Default for DiversifierIndex {
     }
 }
 
+impl From<u32> for DiversifierIndex {
+    fn from(i: u32) -> Self {
+        u64::from(i).into()
+    }
+}
+
+impl From<u64> for DiversifierIndex {
+    fn from(i: u64) -> Self {
+        let mut result = DiversifierIndex([0; 11]);
+        (&mut result.0).copy_from_slice(&i.to_le_bytes());
```

Comment:
This will panic, because [`&[T]::copy_from_slice` requires that `src` and `self` are the same length](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.copy_from_slice), but here `result.0` has length 11, and `i.to_le_bytes()` has length 8. This should be:
```suggestion
        (&mut result.0[..8]).copy_from_slice(&i.to_le_bytes());
```